[NM-62] Add population pyramid plot

krisdoyon opened the pull request:
## Description

Please describe your changes here

## Type of version change

Minor

## Checklist

- [ ] I have incremented version number, or version needs no increment
- [ ] The build passed successfully, or failed because of ADR tests

krisdoyon commented:
/update-snapshots

r-ash reviewed (changes requested):
From a first look, this looks excellent. Couple of small comments I've left you but this is looking really good. Components are impressively tidy for the amount of logic you had to get in there.

Re formatting the tooltips to show +ve I think I can handle that in metadata. We use numeral js http://numeraljs.com/#format formats. This is what `formatOutput` function uses under the hood. So I think I can force those to show +ve by wrapping the current format in brackets.

I also spotted an issue with a few countries, I tried with Kenya, Angola and DRC. If you import Kenya and select any area level higher than the lowest. I get

```
Uncaught (in promise) TypeError: fullParentMap[area_id] is undefined
    aggregatePopulationIndicators filter.ts:305
    aggregatePopulationIndicators filter.ts:303
    _callee7$ filter.ts:325
    Babel 6

```

krisdoyon commented on src/app/static/src/app/components/plots/population/Population.vue:
```diff
@@ -0,0 +1,172 @@
+              return (value * 100).toFixed(0) + "%";
+            } else {
+              return value.toLocaleString(); 
+            }
```
Thanks for this, it's exhibiting the same issue as the tooltip now with negative values on the left side of x-axis, but once we add some kind of absolute value fix it should resolve in both places.

krisdoyon commented on src/app/static/src/app/components/plots/population/PopulationGrid.vue:
```diff
@@ -0,0 +1,95 @@
+
+const subplotsConfig = {
+  columns: 3,
+  rows: 4,
```
Switched to 3x3 grid, increased the aspect ratio slightly which should increase the height of each chart while still keeping them responsive. I agree that this looks better, seems to help with the tooltip too though it's still a little touchy

krisdoyon commented:
The errors you were getting with Kenya, DRC etc. should be fixed now. The way the aggregation logic was written it was making some incorrect assumptions about population data being uploaded only at the most disaggregated level. Modified the filtering logic a bit, and also added a check when aggregating to see if an uploaded indicator already exists. If it does exist, it will now be used directly, and if not it will be built from aggregated child indicators as it was doing previously.

krisdoyon commented:
/update-snapshots

krisdoyon commented:
/update-snapshots

krisdoyon commented:
/update-snapshots

krisdoyon commented:
/update-snapshots

r-ash commented:
/update-snapshots

r-ash commented:
/update-snapshots

r-ash commented:
/update-snapshots
